include/rc_export.h

```diff
@@ -0,0 +1,100 @@
+#ifndef RC_EXPORT_H
+#define RC_EXPORT_H
+
+/* These macros control how callbacks and public functions are defined */
+
+/* RC_SHARED should be defined when building rcheevos as a shared library (e.g. dll/dylib/so). External code should not define this macro. */
+/* RC_STATIC should be defined when building rcheevos as a static library. External code should also define this macro. */
+/* RC_IMPORT should be defined for external code using rcheevos as a shared library. */
+
+/* For compatibility, if none of these three macros are defined, then the build is assumed to be RC_STATIC */
+
+#if !defined(RC_SHARED) && !defined(RC_STATIC) && !defined(RC_IMPORT)
+  #define RC_STATIC
+#endif
+
+#if (defined(RC_SHARED) && defined(RC_STATIC)) || (defined(RC_SHARED) && defined(RC_IMPORT)) || (defined(RC_STATIC) && defined(RC_IMPORT))
+  #error RC_SHARED, RC_STATIC, and RC_IMPORT are mutually exclusive
+#endif
+
+/* RC_CXX_GUARD_BEGIN and RC_CXX_GUARD_END should be used for all headers, to enforce C linkage and the C calling convention */
+/* RC_CXX_GUARD_BEGIN should be placed after #include's and before header declarations */
+/* RC_CXX_GUARD_END should be placed after header declarations */
+
+/* example usage
+ *
+ * #ifndef RC_HEADER_H
+ * #define RC_HEADER_H
+ *
+ * #include <stdint.h>
+ *
+ * RC_CXX_GUARD_BEGIN
```

Comment:
I'm still not sold on using #defines for this. Every library I've looked at just has the raw `#ifndef __cplusplus extern "C" {`

* https://github.com/madler/zlib/blob/643e17b7498d12ab8d15565662880579692f769d/zlib.h#L36-L38
* https://github.com/libsdl-org/SDL/blob/b937c54b665f134f241313eb45a4e0110807ae2c/include/SDL3/SDL_version.h#L34-L37
* https://github.com/openssl/openssl/blob/e4542332fa36eab6d6bbf33815bde433ade3b547/include/openssl/decoder.h#L25-L27

Reply:
https://github.com/mgba-emu/mgba/blob/3a5642fcb81d3c279f8f385828e95ca3029d527d/include/mgba-util/common.h#L9-L15

Some libraries do use a macro for this (well, mGBA is the only one I know on hand)

Reply:
RetroArch apparently also does it https://github.com/libretro/RetroArch/blob/2980eb7e1296498305b9c047c82050ffd4d85778/libretro-common/include/retro_common_api.h#L41-L59

Reply:
> RetroArch apparently also does it 

Not always.
https://github.com/libretro/RetroArch/blob/2980eb7e1296498305b9c047c82050ffd4d85778/libretro-common/include/libretro.h#L30-L32

And definitely not consistently.
```
$ grep -rnI "extern \"C\" {" * | grep -v deps | wc -l
462
$ grep -rnI RETRO_BEGIN_DECLS * | grep -v deps | wc -l
209
```

Reply:
The closest I can find to any sort of industry standard is [this document for writing gnu libraries](https://www.gnu.org/software/libtool/manual/html_node/C-header-files.html). It uses `BEGIN_C_DECLS` and `END_C_DECLS`, which do seem somewhat familiar, and googling for `BEGIN_C_DECLS` returns plenty of hits.

Would you be willing to change the defines to `RC_BEGIN_C_DECLS` and `RC_END_C_DECLS`?

Reply:
I would be willing to do that. I mainly did CXX_GUARD as I only remember mGBA doing this and that's what it used (although in retrospect this is more an exception than the norm)